lib/visual_metrics.py

```diff
@@ -286,57 +287,69 @@ def ListOneTarget(codecs, rate, videofile, do_score, datatable,
     if do_score and not bestsofar.Result():
       bestsofar.Execute()
       bestsofar.Store()
-    assert(bestsofar.Result())
-    # Ignore results that score less than zero.
-    if my_optimizer.Score(bestsofar) < 0.0:
-      return
-    if full_results:
-      # Datatable is a dictionary of codec name -> result sets.
-      # Each result set is an array containing result info.
-      # Each result info is a dictionary containing the
-      # ID of the configuration used, the
-      # target bitrate, the command line, the score and the result.
-      (datatable.setdefault(codec_name, {})
-          .setdefault(videofile.basename, [])
-          .append({'config_id': bestsofar.encoder.Hashname(),
-                   'target_bitrate': rate,
-                   'encode_command': bestsofar.EncodeCommandLine(),
-                   'score': my_optimizer.Score(bestsofar),
-                   'result': bestsofar.ResultWithoutFrameData()}))
-    else:
-      # Datatable is a dictionary of codec name -> result sets.
-      # Each result set is an array containing result info.
-      # Each result info contains the achieved bitrate and the PSNR.
-      (datatable.setdefault(codec_name, {})
-          .setdefault(videofile.basename, [])
-          .append((bestsofar.result['bitrate'], bestsofar.result['psnr'])))
-
-
-def ListMpegResults(codecs, do_score, datatable, score_function=None,
-                    full_results=False):
+    EnterOneTarget(codec_name, my_optimizer, bestsofar, videofile,
+                   datatable)
+
+
+def EnterOneTarget(codec_name, my_optimizer, bestsofar, videofile,
```

Comment:
Is it really the "best so far"? I don't see anything in the calling algorithm that implies this.